### PR TITLE
Fixed support for If-None-Match: * headers for updates.

### PR DESCRIPTION
**Pull request by n1k0**
Currently the api when updating and deleting resources raise when the `safe` option is passed as `true` and no `last_modified` value is provided in the resource object or through the dedicate option.

It appears this only makes sense for deletions, but not for updates where one might want to ensure the resource won't be created automatically if it doesn't exist on the server.

This patch updates the code so it now only raises when asking for safe deletions without providing any `last_modified` information. 

r=? @Natim

### CODE REVIEW

File: test/requests_test.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ chai.should();
 chai.config.includeStack = true;
 
 
-describe("requests module", () => {
+describe.only("requests module", () => {
```
Comment (Natim) [inline test/requests_test.js]: nit: only

File: test/integration_test.js [hunk 2]
```diff
@@ -698,6 +698,15 @@ describe("Integration tests", function() {
...
               });
+
+              it("should not override existing data with no last_modified", () => {
+                return coll.createRecord({title: "foo"})
+                  .then(({data}) => coll.updateRecord({
+                    id: data.id,
+                    title: "foo",
+                  }, {safe: true}))
+                  .should.be.rejectedWith(Error, /412 Precondition Failed/);
+              });
```
Comment (n1k0) [inline test/integration_test.js]: We should add a test to ensure a non-existent resource is created when safe is true.

**Comment by leplatrem** [comment]
r+ !

**Comment by n1k0** [comment]
v0.4.2 released https://github.com/Kinto/kinto-client/releases/tag/v0.4.2

**Pull request merged**